packages/langchain/src/orchestration/util.ts

```diff
@@ -174,3 +200,45 @@ export function mapOutputToChatResult(
     }
   };
 }
+
+/**
+ * Converts orchestration stream chunk to amessage chunk.
+ * @param chunkData - The content of the message.
+ * @param delta - The delta content from the chunk.
+ * @param defaultRole - The default role to use if not specified in the delta.
+ * @returns A message chunk compatible with Langchain's {@link AIMessageChunk}
+ * @internal
+ */
+export function _convertOrchestrationChunkToMessageChunk(
+  chunkData: CompletionPostResponseStreaming,
+  delta: ChatDelta,
+  defaultRole?: string
+): BaseMessageChunk {
+  const { module_results, request_id } = chunkData;
+  const role = delta.role ?? defaultRole ?? 'assistant';
+  const content = delta.content ?? '';
+
+  // Handle additional kwargs for tool calls
+  const additional_kwargs: Record<string, unknown> = {};
+
+  // Handle tool calls
+  if (delta.tool_calls && delta.tool_calls.length > 0) {
+    additional_kwargs.tool_calls = delta.tool_calls;
+  }
+
+  // Create tool call chunks if present
+  let tool_call_chunks: ToolCallChunk[] = [];
+  if (Array.isArray(delta.tool_calls)) {
+    tool_call_chunks = mapOrchestrationToLangchainToolCallChunk(
+      delta.tool_calls
+    );
+  }
+  const toolCallId = delta.tool_calls?.[0]?.id ?? undefined;
+  // Use OrchestrationMessageChunk to represent message chunks for roles like 'tool' and 'user' too
+  return new OrchestrationMessageChunk(
```

Comment:
In other chat model clients in the langchain repo, various other `MessageChunk` like for e.g. `HumanMessageChunk` , `ToolMessageChunk` are created based on roles.

But, in the end the upstream function processes all chunks as `AIMessageChunk`, so I chose to use `OrchestrationMessageChunk` (which extends `AIMessageChunk`) to represent all types of message chunks with some additional properties instead.